File: libs/jetpack/lib-arch/src/main/java/com/yyxnb/what/arch/base/Java8Observer.java
package com.yyxnb.what.arch.base;

import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.LifecycleOwner;

import com.yyxnb.what.core.interfaces.ILifecycle;

import java.io.Serializable;

/**
 * 打印页面生命周期
 *
 * @author yyx
 */
public class Java8Observer implements ILifecycle, Serializable {

    private String tag;

    public Java8Observer(String tag) {
        this.tag = tag;
    }

    @Override
    public void onCreate(@NonNull LifecycleOwner owner) {
        Log.e(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onStart(@NonNull LifecycleOwner owner) {
        Log.w(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onResume(@NonNull LifecycleOwner owner) {
        Log.w(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onPause(@NonNull LifecycleOwner owner) {
        Log.w(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onStop(@NonNull LifecycleOwner owner) {
        Log.w(tag, owner.getLifecycle().getCurrentState().name());
    }

    @Override
    public void onDestroy(@NonNull LifecycleOwner owner) {
        Log.e(tag, owner.getLifecycle().getCurrentState().name());
    }
}

File: arch/src/main/java/com/yyxnb/arch/base/Java8Observer.kt
package com.yyxnb.arch.base

import android.arch.lifecycle.DefaultLifecycleObserver
import android.arch.lifecycle.LifecycleOwner
import android.util.Log
import java.io.Serializable

class Java8Observer(var tag: String) : DefaultLifecycleObserver, Serializable {

    override fun onCreate(owner: LifecycleOwner) {
        Log.w("---onCreate $tag", owner.lifecycle.currentState.name)
    }

    override fun onStart(owner: LifecycleOwner) {
        Log.i("---onStart $tag", owner.lifecycle.currentState.name)
    }

    override fun onResume(owner: LifecycleOwner) {
        Log.i("---onResume $tag", owner.lifecycle.currentState.name)
    }

    override fun onPause(owner: LifecycleOwner) {
        Log.i("---onPause $tag", owner.lifecycle.currentState.name)
    }

    override fun onStop(owner: LifecycleOwner) {
        Log.i("---onStop $tag", owner.lifecycle.currentState.name)
    }

    override fun onDestroy(owner: LifecycleOwner) {
        Log.e("---onDestroy $tag", owner.lifecycle.currentState.name)
    }

}